homeassistant/components/media_source/local_source.py

```diff
@@ -149,19 +180,24 @@ class LocalMediaView(HomeAssistantView):
     Returns media files in config/media.
     """
 
-    url = "/media/{location:.*}"
+    url = "/local_source/{source_dir_id}/{location:.*}"
     name = "media"
 
     def __init__(self, hass: HomeAssistant):
         """Initialize the media view."""
         self.hass = hass
 
-    async def get(self, request: web.Request, location: str) -> web.FileResponse:
+    async def get(
+        self, request: web.Request, source_dir_id: str, location: str
+    ) -> web.FileResponse:
         """Start a GET request."""
         if location != sanitize_path(location):
             return web.HTTPNotFound()
 
-        media_path = Path(self.hass.config.path("media", location))
+        if source_dir_id not in self.hass.config.media_dirs:
+            return web.HTTPNotFound()
+
+        media_path = Path(self.hass.config.media_dirs[source_dir_id], location)
```

Comment:
Let's pass the MediaSource into this view and call `source.async_full_path`. That way we keep resolving source dir ID + location to full path in 1 location. We can update `async_full_path` to return a Path.